### PR TITLE
MOBN-1347 | Add Provision to edit source

### PR DESCRIPTION
**Pull request by supriyatw**
# JIRA TICKET NAME:
[Mobn 1347](https://issues.openmrs.org/browse/<ticket-id>)

# Summary:

### CODE REVIEW

File: src/apps/sources/pages/EditSourcePage.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                        open={Boolean(menuAnchor)}
+                        onClose={handleMenuClose}
+                    >
+                        <MenuItem>
+                            <Link replace className="link" to={sourceUrl}>
+                                Discard changes and view
+                            </Link>
+                        </MenuItem>
+                    </Menu>
+                </>
```
Comment (kirity-mahanty) [inline src/apps/sources/pages/EditSourcePage.tsx]: This component is also used both in Source and Dictionary. We can create it as a common component in Container

File: src/apps/sources/pages/ViewSourcePage.tsx [hunk 2]
```diff
@@ -97,6 +110,15 @@ export const ViewSourcePage: React.FC<Props> = ({
...
         </Grid>
+        {!showEditButton ? null : (
+            <Link to={`${url}edit/`}>
+              <Tooltip title="Edit this Source">
+                <Fab color="primary" className="fab">
+                  <EditIcon />
+                </Fab>
+              </Tooltip>
+            </Link>
+        )}
```
Comment (kirity-mahanty) [inline src/apps/sources/pages/ViewSourcePage.tsx]: This is also common for Source and Dictionary. We can move it to Containers.

Comment (supriyatw) [reply src/apps/sources/pages/ViewSourcePage.tsx]: Made necessary changes please check

File: src/apps/sources/pages/EditSourcePage.tsx (hunk 1 above)
Comment (supriyatw) [reply src/apps/sources/pages/EditSourcePage.tsx]: Please check the changes made.

File: src/apps/concepts/pages/ViewConceptPage.tsx [hunk 3]
```diff
@@ -84,13 +85,7 @@ const ViewConceptPage: React.FC<Props> = ({
...
         </Grid>
         {!showEditButton ? null : (
-          <Link replace to={`${url}edit/?linkedDictionary=${linkedDictionary}`}>
-            <Tooltip title="Edit this concept">
-              <Fab color="primary" className="fab">
-                <EditIcon />
-              </Fab>
-            </Tooltip>
-          </Link>
+            <EditButton url={`${url}edit/?linkedDictionary=${linkedDictionary}`} title="Edit this concept"/>
```
Comment (kirity-mahanty) [inline src/apps/concepts/pages/ViewConceptPage.tsx]: Typo: `c` should be capital in `concept`. For Dictionary and Source, D and S are capital resp.

**Pull request merged**